src/main/java/com/tabnine/statusBar/StatusBarEmptySymbolGenerator.kt

```diff
@@ -1,20 +1,17 @@
-package com.tabnine.statusBar;
-
-import java.util.Arrays;
-import java.util.List;
+package com.tabnine.statusBar
 
 // Workaround to address the issue where IJ does not render the width of the status bar icon
 // correctly when the icon changes
 // because it doesn't detect a change in the widget text (it renders it correctly just after
 // document change occurs).
 // Creating a different empty text each time will solve the problem.
-public class StatusBarEmptySymbolGenerator {
-  private static final String EMPTY_SYMBOL = "\u0000";
-  private static final List<String> EMPTY_SYMBOLS =
-      Arrays.asList(EMPTY_SYMBOL, EMPTY_SYMBOL + EMPTY_SYMBOL);
-  private int currentSymbolIndex = 0;
+class StatusBarEmptySymbolGenerator {
```

Comment:
you don't need a class with a companion object here - you can declare top level static consts and functions in Kotlin - so you just expose a utility function of `getEmptySimbol`

Reply:
the problematic use case here is that two components use this static object, and then the first one changes it, the second one changes it back, and then the first one calls it and gets the same symbol as before which will lead to the component not being rendered.

Reply:
oh so each one maintains an instance - I see.
sababa